src/machines/systemIO/systemIOMachine.ts

```diff
@@ -0,0 +1,448 @@
+      async ({ input: context }: { input: SystemIOContext }) => {
+        const folders: Project[] = []
+        return folders
+      }
+    ),
+    [SystemIOMachineActors.createProject]: fromPromise(
+      async ({
+        input: { context, requestedProjectName },
+      }: {
+        input: { context: SystemIOContext; requestedProjectName: string }
+      }) => {
+        return { message: '', name: '' }
+      }
+    ),
+    [SystemIOMachineActors.deleteProject]: fromPromise(
+      async ({
+        input: { context, requestedProjectName },
+      }: {
+        input: { context: SystemIOContext; requestedProjectName: string }
+      }) => {
+        return { message: '', name: '' }
+      }
+    ),
+    [SystemIOMachineActors.renameProject]: fromPromise(
+      async ({
+        input: { context, requestedProjectName, projectName },
+      }: {
+        input: {
+          context: SystemIOContext
+          requestedProjectName: string
+          projectName: string
+        }
+      }): Promise<{ message: string; newName: string; oldName: string }> => {
+        return { message: '', newName: '', oldName: '' }
+      }
+    ),
+    [SystemIOMachineActors.createKCLFile]: fromPromise(
+      async ({
+        input,
+      }: {
+        input: {
+          context: SystemIOContext
+          requestedProjectName: string
+          requestedFileName: string
+          requestedCode: string
+          rootContext: AppMachineContext
+        }
+      }): Promise<{
+        message: string
+        fileName: string
+        projectName: string
+      }> => {
+        return { message: '', fileName: '', projectName: '' }
+      }
+    ),
+    [SystemIOMachineActors.checkReadWrite]: fromPromise(
+      async ({
+        input: { context, requestedProjectDirectoryPath },
+      }: {
+        input: {
+          context: SystemIOContext
+          requestedProjectDirectoryPath: string
+        }
+      }): Promise<{ value: boolean; error: unknown }> => {
+        return { value: true, error: undefined }
+      }
+    ),
+  },
+}).createMachine({
+  initial: SystemIOMachineStates.idle,
+  // Remember, this machine and change its projectDirectory at any point
+  // '' will be no project directory, aka clear this machine out!
+  // To be the absolute root of someones computer we should take the string of path.resolve() in node.js which is different for each OS
+  context: () => ({
+    folders: [],
+    defaultProjectFolderName: DEFAULT_PROJECT_NAME,
+    projectDirectoryPath: NO_PROJECT_DIRECTORY,
+    hasListedProjects: false,
+    requestedProjectName: { name: NO_PROJECT_DIRECTORY },
+    requestedFileName: {
+      project: NO_PROJECT_DIRECTORY,
+      file: NO_PROJECT_DIRECTORY,
+    },
+    canReadWriteProjectDirectory: { value: true, error: undefined },
+    clearURLParams: { value: false },
+  }),
+  states: {
+    [SystemIOMachineStates.idle]: {
+      on: {
+        // on can be an action
```

Comment:
What does this mean? `on` refers only to events, which then can have actions associated with them, is that what you mean?